zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/remote/RemoteInterpreter.java

```diff
@@ -103,7 +103,7 @@ public String getSessionId() {
   }
 
   public synchronized RemoteInterpreterProcess getOrCreateInterpreterProcess() throws IOException {
-    if (this.interpreterProcess != null) {
+    if (this.interpreterProcess != null && interpreterProcess.isRunning()) {
       return this.interpreterProcess;
     }
     ManagedInterpreterGroup intpGroup = getInterpreterGroup();
```

Comment:
before creating a new one, if `this.interpreterProcess != null` should we close it or something even when it is not running now?

Reply:
The original code is not perfect.
Because `RemoteInterpreter.java::getOrCreateInterpreterProcess()` is called, an available RemoteInterpreter process must be returned.

`interpreterProcess.isRunning()` Will call `RemoteInterpreterUtils.java::checkIfRemoteEndpointAccessible()` Check if the remote interpreter is available through the socket.